artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/QueueImpl.java

```diff
@@ -3458,9 +3460,15 @@ private long calculateRedeliveryDelay(final AddressSettings addressSettings, fin
       long redeliveryDelay = addressSettings.getRedeliveryDelay();
       long maxRedeliveryDelay = addressSettings.getMaxRedeliveryDelay();
       double redeliveryMultiplier = addressSettings.getRedeliveryMultiplier();
+      double collisionAvoidanceFactor = addressSettings.getRedeliveryCollisionAvoidanceFactor();
 
       int tmpDeliveryCount = deliveryCount > 0 ? deliveryCount - 1 : 0;
       long delay = (long) (redeliveryDelay * (Math.pow(redeliveryMultiplier, tmpDeliveryCount)));
+      if (collisionAvoidanceFactor > 0) {
+         Random random = RandomUtil.getRandom();
```

Comment:
RandomUtil return a Thread local random?

Reply:
I'm not sure what you mean. It returns a `static final` instance of `java.util.Random`. Is that what you're asking about?

Reply:
I mean this https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ThreadLocalRandom.html
Random is synchronized on so it would block any concurrent calls

Reply:
Fair enough. I've pushed the change.